### PR TITLE
SONAR-7458 API to get runtime version of SonarQube

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/SonarQubeVersion.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  /**
+   * Constant for version 5.5
+   */
+  public static final Version V5_5 = Version.create(5, 5);
+
+  /**
+   * Constant for version 5.6
+   */
+  public static final Version V5_6 = Version.create(5, 6);
```
Comment (henryju) [inline sonar-plugin-api/src/main/java/org/sonar/api/SonarQubeVersion.java]: Are you sure there will be a v 5.6 ;)

**Comment by henryju** [comment]
LGTM

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Version.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+import static java.lang.Integer.parseInt;
+import static org.apache.commons.lang.StringUtils.substringAfter;
+import static org.apache.commons.lang.StringUtils.substringBefore;
+import static org.apache.commons.lang.StringUtils.trimToEmpty;
+
+/**
+ * TODO document equals/comparison without qualifier
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Version.java]: INFO Complete the task associated to this TODO comment. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1135)

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Version.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    String s = trimToEmpty(text);
+    String qualifier = substringAfter(s, QUALIFIER_SEPARATOR);
+    if (!qualifier.isEmpty()) {
+      s = substringBefore(s, QUALIFIER_SEPARATOR);
+    }
+    List<String> split = SEQUENCE_SPLITTER.splitToList(s);
+    int major = 0;
+    int minor = 0;
+    int patch = 0;
+    if (split.size() > 0) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Version.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1155)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.

File: sonar-plugin-api/src/main/java/org/sonar/api/SonarQubeVersion.java (hunk 1 above)
Comment (simonbrandhof) [inline sonar-plugin-api/src/main/java/org/sonar/api/SonarQubeVersion.java]: There will be ! (it will be the next LTS)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request closed without merging**